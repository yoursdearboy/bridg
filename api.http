@base=http://127.0.0.1:8000

# Spaces

### Index spaces

GET {{base}}/spaces

## Sites

### Index sites in the space

GET {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/sites

## Subjects

### Index subjects in the space

GET {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects

### Show a subject in the space

GET {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects/f0519f08-d22f-40ff-8800-cee82c4e925b

### Create a subject in the space

POST {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects
Content-Type: application/json

{
  "status": "candidate",
  "status_date": "2025-03-01T12:00:00",
  "performing_biologic_entity": {
    "administrative_gender_code": "M",
    "birth_date": "2000-03-01",
    "death_date": null,
    "death_date_estimated_indicator": false,
    "death_indicator": false,
    "name": {
      "use": "official", "family": "Test", "given": "Dude"
    }
  },
  "assigned_study_site_protocol_version_relationship": [
    "6a8e6e2b-9537-408f-bd92-a5b83ad2e750"
  ]
}

### Lookup a subject

POST {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects/lookup
Content-Type: application/json

{
  "performing_biologic_entity": {
    "name": {
      "use": "official", "family": "Test", "given": "Dude"
    }
  }
}

## Persons

### Show a person

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285

### Names

#### Show a person names

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/names

#### Create a person name

POST {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/names
Content-Type: application/json

{
  "use": "official",
  "family": "Works",
  "given": "Good",
  "prefix": "Jr."
}

### Postal addresses

#### Show a person postal addresses

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/postal_addresses

#### Create a person postal address

POST {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/postal_addresses
Content-Type: application/json

{
  "use": "tmp",
  "street": "Pushkina",
  "building": "Kolotushkina",
  "country": "Lukomorje",
  "municipality": "Great tree",
  "state": "",
  "zip": "111111",
}

### Telecommunication addresses

#### Show a person telecommunication addresses

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/telecommunication_addresses

#### Create a person telecommunication address

POST {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/telecommunication_addresses
Content-Type: application/json

{
  "use": "h",
  "shema": "tel",
  "address": "8(800)5553535"
}