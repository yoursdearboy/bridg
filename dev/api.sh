#!/bin/sh

ROOTPATH=$(pwd)

export PYTHONPATH=$ROOTPATH

fastapi dev $ROOTPATH/api/main.py
