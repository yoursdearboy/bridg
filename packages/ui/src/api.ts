import {
  Configuration,
  ObservationResultApi,
  PersonsApi,
  SitesApi,
  DefaultApi as SpacesApi,
  SubjectsApi,
} from "api-ts";

const config = new Configuration({
  basePath: "/api",
});

export const sites = new SitesApi(config);
export const spaces = new SpacesApi(config);
export const subjects = new SubjectsApi(config);
export const persons = new PersonsApi(config);
export const observationResult = new ObservationResultApi(config);

observationResult
  .indexSpacesSpaceIdSubjectsSubjectIdObservationObsIdResultGet({
    spaceId: "ce946229-9746-46cd-8dd3-b27a2fbfd48a",
    subjectId: "1944c046-95b3-4cb4-82e2-c789950e29fc",
    obsId: "8ac78da9-3c23-4770-94e9-ebaee8432b4b",
  })
  .then((x) => {
    console.log(x);
  })
  .catch(console.error);

export default {
  sites,
  spaces,
  subjects,
  persons,
};
