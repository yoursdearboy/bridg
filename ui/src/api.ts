import { DefaultApi as SpacesApi, Configuration, SubjectsApi } from "bridg-ts";

const config = new Configuration({
  basePath: "/api",
});

export const spaces = new SpacesApi(config);
export const subjects = new SubjectsApi(config);

export default {
  spaces,
  subjects,
};
