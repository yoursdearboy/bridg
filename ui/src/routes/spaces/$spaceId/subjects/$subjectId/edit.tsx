// src/routes/spaces/$spaceId/subjects/$subjectId.edit.tsx
import { createFileRoute } from "@tanstack/react-router";
import { Text } from "@mantine/core";

export const Route = createFileRoute(
  "/spaces/$spaceId/subjects/$subjectId/edit"
)({
  component: EditComponent,
  beforeLoad: () => ({
    breadcrumb: "Edit Patient",
  }),
});

function EditComponent() {
  return <Text>Edit Patient </Text>;
}
