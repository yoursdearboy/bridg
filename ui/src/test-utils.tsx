import { MantineProvider } from "@mantine/core";
import { DatesProvider } from "@mantine/dates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  createRootRoute,
  createRoute,
  createRouter,
  RouterProvider,
  type AnyRoute,
} from "@tanstack/react-router";
import { act, render } from "@testing-library/react";
import "./i18n";

// eslint-disable-next-line react-refresh/only-export-components
const App = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient();

  return (
    <QueryClientProvider client={queryClient}>
      <MantineProvider defaultColorScheme="auto">
        <DatesProvider settings={{ locale: "en" }}>{children}</DatesProvider>
      </MantineProvider>
    </QueryClientProvider>
  );
};

export const renderRoute = (
  r: AnyRoute,
  { params }: { params?: object } = {}
) => {
  const rootRoute = createRootRoute();
  const testRoute = createRoute({
    ...r.options,
    params: params,
    getParentRoute: () => rootRoute,
    path: "/",
  });
  const routeTree = rootRoute.addChildren([testRoute]);
  const router = createRouter({
    routeTree,
  });
  return act(() =>
    render(
      <App>
        <RouterProvider router={router} />
      </App>
    )
  );
};
