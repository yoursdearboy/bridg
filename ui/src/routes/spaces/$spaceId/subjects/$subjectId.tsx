// src/routes/spaces/$spaceId/subjects/$subjectId.tsx
import api from "@/api";
import ButtonLink from "@/components/ButtonLink";
import { PersonCard } from "@/components/person/PersonCard";
import { Grid, Group, Stack, Title } from "@mantine/core";
import { createFileRoute } from "@tanstack/react-router";
import type { StudySubject } from "bridg-ts";
import { useTranslation } from "react-i18next";

export const Route = createFileRoute("/spaces/$spaceId/subjects/$subjectId")({
  component: RouteComponent,
  loader: ({ params }) =>
    api.subjects.showSpacesSpaceIdSubjectsSubjectIdGet(params),
  beforeLoad: () => ({
    breadcrumb: ({ loaderData: subject }: { loaderData: StudySubject }) =>
      subject.performingBiologicEntity?.primaryName || "Anonymous subject",
  }),
});

function RouteComponent() {
  const subject = Route.useLoaderData();
  const { t } = useTranslation();

  return (
    <Stack gap="md">
      <Group justify="space-between">
        <Title order={2}>{t("Patient Information")}</Title>
        <ButtonLink to="..">Back to List</ButtonLink>
      </Group>

      <Grid>
        {/* Patient Card (only shown for biologic entities) */}
        {subject.performingBiologicEntity && (
          <Grid.Col span={{ base: 12, md: 6 }}>
            <PersonCard person={subject.performingBiologicEntity} />
          </Grid.Col>
        )}
      </Grid>
    </Stack>
  );
}
