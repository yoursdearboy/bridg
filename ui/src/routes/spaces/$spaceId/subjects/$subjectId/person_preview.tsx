import { createFileRoute } from "@tanstack/react-router";
import { PatientCard } from "@/components/PatientCard";
import { Stack, Text } from "@mantine/core";
import api from "@/api";

export const Route = createFileRoute(
  "/spaces/$spaceId/subjects/$subjectId/person_preview"
)({
  component: EditComponent,
  loader: ({ params }) =>
    api.subjects.showSpacesSpaceIdSubjectsSubjectIdGet(params),
});

function EditComponent() {
  const { spaceId, subjectId } = Route.useParams();
  const subject = Route.useLoaderData();

  return (
    <Stack>
      <PatientCard subject={subject} />
      <Text>
        Editing subject {subjectId} in space {spaceId}
      </Text>
    </Stack>
  );
}
