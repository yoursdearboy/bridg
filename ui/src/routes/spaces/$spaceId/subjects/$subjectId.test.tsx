import api from "@/api";
import { renderRoute } from "@/test-utils";
import dayjs from "dayjs";
import { expect, it, vi } from "vitest";
import { Route } from "./$subjectId";

it("show page renders correctly", async () => {
  vi.spyOn(
    api.subjects,
    "showSpacesSpaceIdSubjectsSubjectIdGet"
  ).mockResolvedValue({
    id: "2703d0bc-7ed4-497c-91c6-30d86a8eb630",
    status: "eligible",
    statusDate: dayjs("2024-11-06T12:00:00").toDate(),
    performingBiologicEntity: {
      id: "8498d20d-6c84-4f09-96c5-5af3a557b1e3",
      administrativeGenderCode: "M",
      birthDate: dayjs("1991-01-01").toDate(),
      deathDate: null,
      deathDateEstimatedIndicator: null,
      deathIndicator: false,
      primaryName: "Donald Trump Jr",
    },
    performingOrganization: null,
  });
  expect((await renderRoute(Route)).asFragment()).toMatchSnapshot();
});
