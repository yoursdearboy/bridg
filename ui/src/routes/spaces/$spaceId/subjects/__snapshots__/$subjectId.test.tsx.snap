// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`show page renders correctly 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
  >
    <div
      class="m_4081bf90 mantine-Group-root"
      style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
    >
      <h2
        class="m_8a5d1357 mantine-Title-root"
        data-order="2"
        style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
      >
        Patient Information
      </h2>
      <button
        aria-current="page"
        class="mantine-focus-auto mantine-active active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-status="active"
        href="/"
        style="--button-color: var(--mantine-color-white);"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Back to List
          </span>
        </span>
      </button>
    </div>
    <style
      data-mantine-styles="inline"
    >
      .__m__-«r5»{--grid-gutter:var(--mantine-spacing-md);}
    </style>
    <div
      class="m_410352e9 mantine-Grid-root __m__-«r5»"
    >
      <div
        class="m_dee7bd2f mantine-Grid-inner"
      >
        <style
          data-mantine-styles="inline"
        >
          .__m__-«r7»{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 62em){.__m__-«r7»{--col-flex-grow:auto;--col-flex-basis:50%;--col-max-width:50%;}}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-«r7»"
        >
          <div
            class="m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
            data-with-border="true"
            style="--paper-radius: var(--mantine-radius-md); --paper-shadow: var(--mantine-shadow-sm); --card-padding: var(--mantine-spacing-lg);"
          >
            <div
              class="m_6d731127 mantine-Stack-root"
              style="--stack-gap: var(--mantine-spacing-sm); --stack-align: stretch; --stack-justify: flex-start;"
            >
              <div
                class="m_4081bf90 mantine-Group-root"
                style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
              >
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                  style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
                >
                  Full Name:
                </p>
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                >
                  Donald Trump Jr
                </p>
              </div>
              <div
                class="m_4081bf90 mantine-Group-root"
                style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
              >
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                  style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
                >
                  Gender Name:
                </p>
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                >
                  M
                </p>
              </div>
              <div
                class="m_4081bf90 mantine-Group-root"
                style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
              >
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                  style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
                >
                  Date of Birth:
                </p>
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                >
                  1&#x2F;1&#x2F;1991
                </p>
              </div>
              <div
                class="m_4081bf90 mantine-Group-root"
                style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
              >
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                  style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
                >
                  Age:
                </p>
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                >
                  34 years
                </p>
              </div>
              <div
                class="m_4081bf90 mantine-Group-root"
                style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
              >
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                  style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
                >
                  Date of Death:
                </p>
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                >
                  Not deceased
                </p>
              </div>
            </div>
          </div>
          <button
            aria-current="page"
            class="mantine-focus-auto mantine-active active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
            data-status="active"
            data-variant="light"
            href="/"
            style="--button-bg: var(--mantine-color-blue-light); --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; margin-top: var(--mantine-spacing-md);"
            type="button"
          >
            <span
              class="m_80f1301b mantine-Button-inner"
            >
              <span
                class="m_811560b9 mantine-Button-label"
              >
                Go to person page
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
